File: solutions/fable-1/fable/python/transpose.py
import sys


def main():
    data = sys.stdin.buffer.read().split()
    if not data:
        return
    r, c = int(data[0]), int(data[1])
    vals = data[2:2 + r * c]
    out = []
    for j in range(c):
        out.append(b" ".join(vals[j::c]))
    sys.stdout.buffer.write(b"\n".join(out) + (b"\n" if out else b""))


if __name__ == "__main__":
    main()

File: solutions/fable-1/fable/ruby/transpose.rb
data = STDIN.read.split
r = data[0].to_i
c = data[1].to_i
vals = data[2, r * c] || []
out = []
c.times do |j|
  row = Array.new(r) { |i| vals[i * c + j] }
  out << row.join(' ')
end
puts out.join("\n") unless out.empty?
